commit d88a159da42a75dbd46ea4f6f9c8059975dab5e8
Author: Mark Johnston <markj@FreeBSD.org>
Date:   Thu Apr 16 19:57:36 2026 +0000

    so_splice: Fix the KTLS check for the sink socket in so_splice()
    
    Reviewed by:    gallatin
    Reported by:    Claude Sonnet 4.6
    Fixes:          1000cc4a0d39 ("so_splice: Disallow splicing with KTLS-enabled sockets")
    MFC after:      1 week
    Differential Revision:  https://reviews.freebsd.org/D56385
---
 sys/kern/uipc_socket.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sys/kern/uipc_socket.c b/sys/kern/uipc_socket.c
index 45290c29f62..3debec547a8 100644
--- a/sys/kern/uipc_socket.c
+++ b/sys/kern/uipc_socket.c
@@ -1768,7 +1768,7 @@ so_splice(struct socket *so, struct socket *so2, struct splice *splice)
 		return (error);
 	}
 	SOCK_SENDBUF_LOCK(so2);
-	if (so->so_snd.sb_tls_info != NULL) {
+	if (so2->so_snd.sb_tls_info != NULL) {
 		SOCK_SENDBUF_UNLOCK(so2);
 		SOCK_UNLOCK(so2);
 		mtx_lock(&sp->mtx);
